### PR TITLE
Added function "select_scatter"

### LINKED ISSUE #21010: select_scatter

### PR DESCRIPTION
**Pull request by sanju4725**
Close #21010

### CODE REVIEW

File: .github/workflows/python-app.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
+    - name: Lint with flake8
+      run: |
+        # stop the build if there are Python syntax errors or undefined names
+        flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics
+        # exit-zero treats all errors as warnings. The GitHub editor is 127 chars wide
+        flake8 . --count --exit-zero --max-complexity=10 --max-line-length=127 --statistics
+    - name: Test with pytest
+      run: |
+        pytest
```
Comment (xoiga123) [inline .github/workflows/python-app.yml:39]: Why did you add this :thinking:?

**Review by xoiga123 — CHANGES_REQUESTED** [review]
Hi @sanju4725! Sorry for the long reply. Please add a test for `test_torch_select_scatter` as well :smile:

File: .github/workflows/python-app.yml (hunk 1 above)
Comment (sanju4725) [reply .github/workflows/python-app.yml]: I just deleted it

File: ivy/functional/frontends/torch/indexing_slicing_joining_mutating_ops.py [hunk 2]
```diff
@@ -360,9 +360,16 @@ def narrow(input, dim, start, length):
     return input[tuple(slices)]
 
 
+@to_ivy_arrays_and_back
+def select_scatter(input, src, dim, index):
+    output = ivy.zeros_like(input)
+    output[dim, index] = src
+    return output
+
```
Comment (hmahmood24) [inline ivy/functional/frontends/torch/indexing_slicing_joining_mutating_ops.py:368]: Hey @sanju4725, you'll also need to fix lint/formatting issues in your implementation. Try using `pre-commit` for auto-formatting and running any formatting checks before committing your work. Thanks

**Review by hmahmood24 — CHANGES_REQUESTED** [review]

File: ivy/functional/frontends/torch/indexing_slicing_joining_mutating_ops.py (hunk 2 above)
Comment (sanju4725) [reply ivy/functional/frontends/torch/indexing_slicing_joining_mutating_ops.py:368]: I have made the necessary changes sir
Please review it again

File: ivy_tests/test_ivy/test_frontends/test_torch/test_indexing_slicing_joining_mutating_ops.py [hunk 3]
```diff
@@ -1593,3 +1593,26 @@ def test_torch_select(
         dim=axis,
         index=idx,
     )
+
```
Comment (hmahmood24) [inline ivy_tests/test_ivy/test_frontends/test_torch/test_indexing_slicing_joining_mutating_ops.py:1596]: @sanju4725 You don't have any input strategies defined for the test using the `@handle_frontend_test` decorator? See other tests in this file for reference

**Review by hmahmood24 — CHANGES_REQUESTED** [review]

File: ivy_tests/test_ivy/test_frontends/test_torch/test_indexing_slicing_joining_mutating_ops.py (hunk 3 above)
Comment (hmahmood24) [reply ivy_tests/test_ivy/test_frontends/test_torch/test_indexing_slicing_joining_mutating_ops.py:1596]: @sanju4725 I don't see any changes with respect to my comment above

Comment (sanju4725) [reply ivy_tests/test_ivy/test_frontends/test_torch/test_indexing_slicing_joining_mutating_ops.py:1596]: sorry sir,
 was unavailable for the past few days
 Ill get back right away

File: ivy/functional/frontends/torch/indexing_slicing_joining_mutating_ops.py (hunk 2 above)
Comment (hmahmood24) [reply ivy/functional/frontends/torch/indexing_slicing_joining_mutating_ops.py:370]: @sanju4725 Ask for a re-review once you've done making all the requested changes on this PR. Thanks

**Comment by hmahmood24** [comment]
Closing this PR as there has been no activity for more than a week.
